app/src/main/java/com/github/se/polyfit/data/processor/LocalDataProcessor.kt

```diff
@@ -0,0 +1,40 @@
+package com.github.se.polyfit.data.processor
+
+import com.github.se.polyfit.data.local.dao.MealDao
+import com.github.se.polyfit.model.meal.MealOccasion
+import java.time.LocalDate
+import javax.inject.Inject
+
+data class DailyCalorieSummary(val date: LocalDate, val totalCalories: Double)
+
+class LocalDataProcessor @Inject constructor(private val mealDao: MealDao) {
+
+  private fun calculateCalories(sinceDate: LocalDate): List<DailyCalorieSummary> {
```

Comment:
To accomodate the other change suggested above, perhaps this can have an optional parameter to calculate on the date given only or since the date given